haystack_experimental/components/retrievers/auto_merging_retriever.py

```diff
@@ -117,41 +117,58 @@ def run(self, matched_leaf_documents: List[Document]):
         """
         Run the AutoMergingRetriever.
 
-        Groups the matched leaf documents by their parent documents and returns the parent documents if the number of
-        matched leaf documents below the same parent is higher than the defined threshold. Otherwise, returns the
-        matched leaf documents.
+        Recursively groups documents by their parents and merges them if they meet the threshold,
+        continuing up the hierarchy until no more merges are possible.
 
         :param matched_leaf_documents: List of leaf documents that were matched by a retriever
         :returns:
-            List of parent documents or matched leaf documents based on the threshold value
+            List of documents (could be a mix of different hierarchy levels) based on threshold values
```

Comment:
There is still only a single threshold value, right? I wonder why it's `threshold values` here then? Are you referring to merge scores here?

Reply:
typo - I've just removed that last part, not need for it